x-pack/test/functional/apps/lens/open_in_lens/agg_based/goal.ts

```diff
@@ -40,8 +40,9 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
     it('should convert to Lens', async () => {
       await visualize.navigateToLensFromAnotherVisulization();
       await lens.waitForVisualization('mtrVis');
-      expect((await lens.getMetricVisualizationData()).length).to.be.equal(1);
-      expect(await lens.getMetricVisualizationData()).to.eql([
+      const data = await lens.getMetricVisualizationData();
```

Comment:
Good work. I noticed that this is a surprisingly expensive operation.

Reply:
Yes, definitely) I was very surprised.

Reply:
@andrewctate, could you, please, suggest some alternatives?

Reply:
No, this is good. I think minimizing how many times it gets called is the best I know how to do. Thank you for doing this!